apps/dcos_dns/src/dcos_dns.erl

```diff
@@ -24,20 +19,62 @@ family(IP) when size(IP) == 8 ->
 
 -spec(get_leader_addr() -> {ok, inet:ip_address()} | {error, term()}).
 get_leader_addr() ->
-    resolve_mesos("leader.mesos").
+    case resolve(<<"leader.mesos">>, inet) of
+        {ok, [IP]} ->
+            {ok, IP};
+        {ok, _IPs} ->
+            {error, not_found};
+        {error, Error} ->
+            {error, Error}
+    end.
+
+-spec(resolve(binary()) -> {ok, [inet:ip_address()]} | {error, term()}).
+resolve(DNSName) ->
+    resolve(DNSName, inet).
+
+-spec(resolve(binary(), inet | inet6) ->
+    {ok, [inet:ip_address()]} | {error, term()}).
+resolve(DNSName, Family) ->
+    resolve(DNSName, Family, 5000).
+
+-spec(resolve(binary(), inet | inet6, timeout()) ->
+    {ok, [inet:ip_address()]} | {error, term()}).
+resolve(DNSName, Family, Timeout) ->
+    DNSNameStr = binary_to_list(DNSName),
+    ParseFun =
+        case Family of
+            inet -> fun inet:parse_ipv4_address/1;
+            inet6 -> fun inet:parse_ipv6_address/1
+        end,
+    case ParseFun(DNSNameStr) of
+        {ok, IP} ->
+            {ok, [IP]};
+        {error, einval} ->
+            imp_resolve(DNSName, Family, Timeout)
+    end.
 
--spec(resolve_mesos(inet_res:dns_name()) ->
+-spec(imp_resolve(binary(), inet | inet6, timeout()) ->
     {ok, inet:ip_address()} | {error, term()}).
-resolve_mesos(DNSName) ->
-    Opts = [{nameservers, dcos_dns_config:mesos_resolvers()}],
-    case inet_res:resolve(DNSName, in, a, Opts, 1000) of
-        {ok, DnsMsg} ->
-            case inet_dns:msg(DnsMsg, anlist) of
-                [DnsRecord] ->
-                    {ok, inet_dns:rr(DnsRecord, data)};
-                _DnsRecords ->
-                    {error, not_found}
+imp_resolve(<<"localhost">>, inet, _Timeout) ->
```

Comment:
What about having the `inet6` case for `localhost`?

Reply:
There is no ipv6 address for `localhost` in dcos-dns:

https://github.com/dcos/dcos-net/blob/master/apps/dcos_dns/src/dcos_dns_sup.erl#L98-L110